utils/genesis/src/lib.rs

```diff
@@ -70,6 +77,15 @@ where
     Ok(network_name)
 }
 
+pub async fn get_network_name_from_genesis_cid(cid: &str) -> Result<String, anyhow::Error> {
```

Comment:
Same comment as for `get_actual_network_name`; it doesn't seem we need to allocate on every call unless I am missing something.

Reply:
not needed, fixed